library/core/src/slice/iter/macros.rs

```diff
@@ -437,6 +447,7 @@ macro_rules! iterator {
 
         impl<'a, T> UncheckedIterator for $name<'a, T> {
             #[inline]
+            #[safety::requires(!is_empty!(self))]
```

Comment:
Doesn't this require a `modifies` clause?

Reply:
Indeed. I am not able to verify this contract yet due to https://github.com/model-checking/kani/issues/1997